### ISSUE TITLE
What happens in the case that a critical bug is found?

### ISSUE THREAD

**Issue by lsaether** [convergentcx/liquid#1]
_What happens if we find a bug in the contracts that we are unable to upgrade out of? What happens if the upgradeability is not robust enough and funds become locked or a bank run occurs?_

We should have some way to do a global shutdown of the system. Until we know for sure that the contracts work how we want them to and will continue to work how we want them to indefinitely we need to have this safeguard. A Convergent owned account (probably the DAO) will be the admin able to withdraw all funds from contracts and either 1) return them in equal percentages to all token holders or 2) put the funds in a new contract.

**Comment by acrdlph** [comment]
I agree - this is a necessary safeguard for the early days that can be removed later on
